docs/gpu.ipynb

```diff
@@ -0,0 +1,344 @@
+{
```

Comment:
<p>I usually feed stuff back to clang, but I guess llc should work as well</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xdslproject/xdsl/pull/514/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Reply:
<p>Mh, I think this is what I'm doing somewhere else too.</p><p>I'll try to use clang everywhere.</p>

Reply:
Oh yeah, oh no!

Are there "GPU simulators" that might be easier to run from within jupyter and that would at least show that stuff is correct...

Reply:
I have no clue yet!
But for clarification, Jupyter is not the issue (I did run all that on my local Jupyter), the docker image _might_ be. It has been a long time I didn't try this kind of things in there though, maybe it's easy now.
But even if it i feasible, getting MLIR to work well with the proper CUDA libraries in there will be fun.. I'll investigate a bit but I'm not sure I'll have the motivation right away, maybe there's more urgent work to do in that area.

Reply:
I am expecting a crash of the CI once we make this a PR. If that happens, I think we can also have a `experimental` folder for jupyter notebooks that are not run on the CI.

Reply:
I agree, I don't quite like hard coded testing. I really would prefer it to be transparent that this is not run/kept uptodate.